packages/block-editor/src/components/link-control/index.js

```diff
@@ -210,8 +210,7 @@ function LinkControl( {
 		// is still unassigned after calling `onChange`.
 		const hadFocusLoss =
 			isEndingEditWithFocus.current &&
-			wrapperNode.current &&
-			! wrapperNode.current.contains( document.activeElement );
+			! wrapperNode.current?.contains( document.activeElement );
```

Comment:
I recall reflecting on this at https://github.com/WordPress/gutenberg/pull/19931#discussion_r372387834 , and observing that this sort of change will be a very common pitfall, and not an equivalent behavior to what would have existed previously.

Which is: If `wrapperNode.current` is undefined, then the negation would yield a value of `true`, which is not what we expect.

I'd recommend to just leave this change out.

I think we could use it here though, in the same file:

https://github.com/WordPress/gutenberg/blob/2cf0a3d8ce079c0a51d409c051c0c053a88beea3/packages/block-editor/src/components/link-control/index.js#L330-L331

Reply:
`! wrapperNode.current?.contains()` === `! (wrapperNode.current && wrapperNode.current.contains()` !== `wrapperNode.current && ! wrapperNode.current.contains()`.

As you said in the comment, we need a rule, `no-optional-chaining-with-negation`.